rundeckapp/grails-spa/packages/ui-trellis/src/app/components/readme-motd/EditProjectFile.vue

```diff
@@ -0,0 +1,161 @@
+                      <summary>
+                        <span v-html="$t('file.readme.help.markdown')"></span>
+                        <span class="more-indicator-verbiage more-info-icon"><i class="glyphicon glyphicon-chevron-right "></i></span>
+                        <span class="less-indicator-verbiage more-info-icon"><i class="glyphicon glyphicon-chevron-down "></i></span>
+                      </summary>
+                      <span v-html="$t('file.readme.help.html')"></span>
+                    </details>
+                  </div>
+                  <ace-editor v-model="fileText"
+                              :soft-wrap-control="true"
+                              height="250"
+                              width="100%"
+                              lang="markdown"
+                              :readOnly="false"
+                  />
+                </div>
+                <div class="card-footer">
+                  <button type="button" class="btn btn-default reset_page_confirm" @click="createProjectHomeLink">Cancel</button>
+                  <button type="submit" class="btn btn-cta reset_page_confirm" @click="saveProjectFile">Save</button>
+                  <template v-if="displayConfig.includes('none')">
+                    <span class="text-warning text-right">
+                      <template v-if="authAdmin">
+                        {{ $t("file.warning.not.displayed.admin.message") }}
+                        <a :href="createProjectConfigureLink"> {{ $t("project.configuration.label") }} </a>
+                      </template>
+                      <template v-else>
+                        {{ $t("file.warning.not.displayed.nonadmin.message") }}
+                      </template>
+                    </span>
+                  </template>
+                </div>
+              </div>
+            </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import {defineComponent} from "vue";
+import AceEditorVue from "@/library/components/utils/AceEditorVue.vue";
+import AceEditor from "@/library/components/utils/AceEditor.vue";
+import {getRundeckContext} from "@/library";
+import {url} from  "@/library/rundeckService"
+import {Notification} from "uiv";
+
+const rundeckClient = getRundeckContext().rundeckClient
+export default defineComponent({
+  name: "EditProjectFile",
+  components: {AceEditor, AceEditorVue},
+  data() {
+    return {
+      fileText: '',
+      markdownSectionOpen: false,
+      errorMsg: ''
+    };
+  },
+  props: {
+    filename: '',
+    displayConfig: [],
+    project: '',
+    authAdmin: false,
+  },
+  computed: {
+    createProjectConfigureLink() {
+      return url('project/' + this.project + '/configure').href
+    },
+  },
+  mounted() {
+    this.getFileText()
+  },
+  methods: {
+    async saveProjectFile() {
+      if(this.fileText === ''){
+        this.fileText = '#This is a test'
+      }
+      const resp = await rundeckClient.sendRequest({
```

Comment:
nitpick (non-blocking): it would be good to move the sendRequest calls to its own service file, the benefits are:

- better separation of concerns, because then the saveProjectFile method focuses on what is relevant for the component, which is handling the error or success instead of the status of the response;
- makes it easier in the future to add unit tests;